src/sentry/ingest/consumer/simple_event.py

```diff
@@ -29,26 +30,43 @@ def process_simple_event_message(
       `symbolicate_event` or `process_event`.
     """
 
-    raw_payload = raw_message.payload.value
-    metrics.distribution(
-        "ingest_consumer.payload_size",
-        len(raw_payload),
-        tags={"consumer": consumer_type},
-        unit="byte",
-    )
-    message: IngestMessage = msgpack.unpackb(raw_payload, use_list=False)
+    try:
+        raw_payload = raw_message.payload.value
+        metrics.distribution(
+            "ingest_consumer.payload_size",
+            len(raw_payload),
+            tags={"consumer": consumer_type},
+            unit="byte",
+        )
+        message: IngestMessage = msgpack.unpackb(raw_payload, use_list=False)
 
-    message_type = message["type"]
-    project_id = message["project_id"]
+        message_type = message["type"]
+        project_id = message["project_id"]
 
-    if message_type != "event":
-        raise ValueError(f"Unsupported message type: {message_type}")
+        if message_type != "event":
+            raise ValueError(f"Unsupported message type: {message_type}")
 
-    try:
-        with metrics.timer("ingest_consumer.fetch_project"):
-            project = Project.objects.get_from_cache(id=project_id)
-    except Project.DoesNotExist:
-        logger.exception("Project for ingested event does not exist: %s", project_id)
-        return
+        try:
+            try:
+                with metrics.timer("ingest_consumer.fetch_project"):
+                    project = Project.objects.get_from_cache(id=project_id)
+            except Project.DoesNotExist:
+                logger.exception("Project for ingested event does not exist: %s", project_id)
+                return
+
+            return process_event(message, project, reprocess_only_stuck_events)
+        except Exception as exc:
+            raise Retriable(exc)
 
-    return process_event(message, project, reprocess_only_stuck_events)
+    except Exception as exc:
+        # Non retriable exceptions raise InvalidMessage, which Arroyo will DLQ.
```

Comment:
not sure this makes a ton of sense. why put retrieable errors in the DLQ if nobody can retry those DLQ items?

Reply:
It's the non-retriable errors not retriable ones that go in the DLQ, so the consumer does not get stuck on those. The retriable errors are things that might be caused by network issues, temporary unavailability of database, etc and should not be DLQed.

Reply:
that's what I mean. IMO retriable errors should go into the DLQ as well. the DLQ is meant to be replayed.

Reply:
I think in an ideal world we'd do that eventually but I'm not sure if we're there yet, and we haven't tried it anywhere. IMO inspection of events and replaying needs to be built in a really fast and easy manner before taking this step.

Today DLQing everything in the case of temporary blips might cause a longer and more manual recovery period and comes with it's own risks that I haven't fully thought through yet.

Let's do it as a follow up later.

Reply:
Would INC-660 messages have been marked as `Retriable` or `InvalidMessage` in this case?

Reply:
since the message was completely empty, it would've failed on the first line and raised `InvalidMessage` and been DLQ.  But more subtlely wrong messages might still slip through the cracks in the future. Note: this is really just a starting point to wire up a DLQ on basic failures, but we will probably want to tweak exactly what gets DLQed later.

Reply:
Sounds good. I think I agree with @untitaker that generally any message that is failing should go into the DLQ, or we risk blocking up the pipeline again because of some unforeseen message or parsing issue. Perhaps with some built in retry and then DLQ after 3 failures or something. 

Building the tools to be able to reprocess easily from the DLQ sounds important too. 😊

Reply:
Are you sure you want to do this in this PR though, before the other tooling is built? I'm happy to change it now but be aware that SRE will potentially have to do large amount of manual consuming from DLQ even in the case of a 30 second temporary unavailability or we risk dropping messages if we experience some very fleeting network or availability issue

Reply:
In the meantime, I'm going to work on defining a schema for this topic so we can use that to determine what is valid or not.

Reply:
No, what you have here seems good to start with, just that we should continue thinking about how to turn it into a general purpose failure handling tool.

Reply:
During INC-660, we crashed inside `process_event`, here: https://github.com/getsentry/sentry/blob/270094f47c840b6f7c084f5e569d3d61d9fda0e2/src/sentry/ingest/consumer/processors.py#L129

See [this sentry error](https://sentry.my.sentry.io/organizations/sentry/issues/683565/events/b1682b6ab8af4abda4fc2a11083b9a2e/).

The kafka message itself was populated, but the event JSON inside was empty.

@lynnagara How often does the consumer currently crash because of temporary network outages? If it's rare enough (say, once per month), I would personally go ahead and DLQ all exceptions.

I do believe this PR is an improvement (any DLQ is better than none), just pointing out that it would not have helped with INC-660.

Reply:
Ok, I realised it's going to be a lot of work to figure out what's retriable and what isn't on a consumer by consumer basis. So I rewrote this to check against the schema whenever an exception is raised to decide whether to DLQ it or not. Since there wasn't a schema for ingest-events, I added one here. https://github.com/getsentry/sentry-kafka-schemas/pull/230